crates/hyperdrive-math/src/short/open.rs

```diff
@@ -157,14 +157,26 @@ impl State {
     /// will pay and comparing this to the amount of base the short will receive
     /// if the variable rate stays the same. The implied rate is just the ROI
     /// if the variable rate stays the same.
+    ///
+    /// To do this, we must adjust the variable rate $r_{adjusted}$ according to
+    /// the position duration $t$ and the variable yield source's compounding
+    /// frequency $f$. The adjusted rate will be:
+    ///
+    /// $$
+    /// r_{adjusted} = ((1 + r_{variable})^{1/f})^{t*f}-1
+    /// $$
     pub fn calculate_implied_rate(
         &self,
         bond_amount: FixedPoint,
         open_vault_share_price: FixedPoint,
         variable_apy: FixedPoint,
+        compounding_frequency: FixedPoint,
     ) -> Result<I256> {
         let base_paid = self.calculate_open_short(bond_amount, open_vault_share_price)?;
-        let base_proceeds = bond_amount * variable_apy;
+        let base_proceeds = bond_amount
+            * (((fixed!(1e18) + variable_apy).pow(fixed!(1e18) / compounding_frequency))
```

Comment:
Shouldn’t we divide the variable apy by the compounding interval in the interior?

Reply:
Actually, nvm. I need to go through this in the whiteboard to see how to get the rate given the apy